src/libs_3rdparty/QSpec/src/primitives/GTWidget.cpp

```diff
@@ -242,6 +244,24 @@ QPoint GTWidget::getWidgetCenter(QWidget* widget) {
     return widget->mapToGlobal(widget->rect().center());
 }
 
+QPoint GTWidget::getWidgetVisibleCenter(QWidget *widget) {
+    return widget->mapFromGlobal(getWidgetVisibleCenterGlobal(widget));
+}
+
+QPoint GTWidget::getWidgetVisibleCenterGlobal(QWidget* widget) {
+    QRect rect = widget->rect();
+    QRect gRect(widget->mapToGlobal(rect.topLeft()), rect.size());
```

Comment:
1. Why do you have 'visible' in the method name? The rect can be not visible here at all?
2. Why do you intersect with a parent rect? Usually parent always contains child. If some parent does not contain the child the intersection may be 0 while both widgets are visible.

Reply:
'Visible' means visible for user (for example she can click this point by mouse)
The widget can has large heigt, so the center will be out of the screen/parent boundaries